bundles/org.openhab.core.config.discovery/src/test/java/org/openhab/core/config/discovery/DiscoveryResultBuilderTest.java

```diff
@@ -0,0 +1,104 @@
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+
+/**
+ * Tests the {@link DiscoveryResultBuilder}.
+ *
+ * @author Christoph Weitkamp - Initial contribution
+ */
+public class DiscoveryResultBuilderTest {
+
+    private static final String BINDING_ID = "bindingId";
+    private static final ThingUID BRIDGE_UID = new ThingUID(new ThingTypeUID(BINDING_ID, "bridgeTypeId"), "bridgeId");
+    private static final ThingTypeUID THING_TYPE_UID = new ThingTypeUID(BINDING_ID, "thingTypeId");
+    private static final ThingUID THING_UID = new ThingUID(THING_TYPE_UID, BRIDGE_UID, "thingId");
+    private static final String KEY1 = "key1";
+    private static final String KEY2 = "key2";
+    private static final String VALUE1 = "value1";
+    private static final String VALUE2 = "value2";
+    private final Map<String, Object> properties = new HashMap<String, Object>() {
+        private static final long serialVersionUID = 1L;
+        {
+            put(KEY1, VALUE1);
+            put(KEY2, VALUE2);
+        }
+    };
+    private DiscoveryResultBuilder builder;
+    private DiscoveryResult discoveryResult;
+
+    @Before
+    public void setup() {
+        builder = DiscoveryResultBuilder.create(THING_UID).withThingType(THING_TYPE_UID).withProperties(properties)
+                .withRepresentationProperty(KEY1).withLabel("Test");
+        discoveryResult = builder.build();
+    }
+
+    @Test
+    public void testDiscoveryResultBuilder() {
+        assertThat(discoveryResult.getThingUID(), is(THING_UID));
+        assertThat(discoveryResult.getThingTypeUID(), is(THING_TYPE_UID));
+        assertThat(discoveryResult.getBindingId(), is(BINDING_ID));
+        assertThat(discoveryResult.getLabel(), is("Test"));
+        assertThat(discoveryResult.getProperties().size(), is(2));
+        assertThat(discoveryResult.getProperties(), hasEntry(KEY1, VALUE1));
+        assertThat(discoveryResult.getProperties(), hasEntry(KEY2, VALUE2));
+        assertThat(discoveryResult.getRepresentationProperty(), is(KEY1));
+        assertThat(discoveryResult.getTimeToLive(), is(DiscoveryResult.TTL_UNLIMITED));
+    }
+
+    @Test
+    public void testDiscoveryResultBuilderWithTTL() {
+        DiscoveryResult otherDiscoveryResult = builder.withTTL(100L).build();
+
+        assertThat(otherDiscoveryResult.getTimeToLive(), is(100L));
+    }
+
+    @Test
+    public void testDiscoveryResultBuilderWithMatchingBridge() {
+        DiscoveryResult otherDiscoveryResult = builder.withBridge(BRIDGE_UID).build();
+
+        assertThat(otherDiscoveryResult.getBridgeUID(), is(BRIDGE_UID));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testDiscoveryResultBuilderWithBridge() {
+        @SuppressWarnings("unused")
+        DiscoveryResult otherDiscoveryResult = DiscoveryResultBuilder
+                .create(new ThingUID(THING_TYPE_UID, "otherThingId")).withBridge(BRIDGE_UID).build();
+    }
+
+    @Test
+    @Ignore
```

Comment:
Why is this test ignored?

Reply:
To streamline our builders (following https://github.com/eclipse-archived/smarthome/issues/5929) I first changes the behavior of the `withProperties(Map<String, Object>)` method to replace existing properties map by the one set. This has been reverted after https://github.com/openhab/openhab-core/pull/1481#discussion_r429679569 and thus the test is failing now. Will come up with a follow-up PR if the change should be applied at all to separate it from this feature.

Reply:
Have a look [here](https://github.com/openhab/openhab-core/commit/59a59b4a0a127a741054beb9ee48bd47735cb5cb) for the changes I am talking about.